execution_engine/src/storage/global_state/in_memory.rs

```diff
@@ -28,18 +30,20 @@ use crate::storage::{
     },
 };
 
+/// Global state implemented purely in memory only. No state is saved to disk, and this is mostly
+/// used for testing purposes.
 pub struct InMemoryGlobalState {
-    pub environment: Arc<InMemoryEnvironment>,
-    pub trie_store: Arc<InMemoryTrieStore>,
-    pub protocol_data_store: Arc<InMemoryProtocolDataStore>,
-    pub empty_root_hash: Blake2bHash,
+    pub(crate) environment: Arc<InMemoryEnvironment>,
+    pub(crate) trie_store: Arc<InMemoryTrieStore>,
+    pub(crate) protocol_data_store: Arc<InMemoryProtocolDataStore>,
+    pub(crate) empty_root_hash: Blake2bHash,
```

Comment:
Recently when I was doing this sort of thing, I just did `pub(crate)` for the whole file...

Reply:
Looking below, I see now that while these structs are `pub`, we are restricting access to these internal details to just this crate, which makes sense.